### PR TITLE
readerwriterqueue: conan v2 support

### PR DESCRIPTION
**Pull request by kletoz**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/readerwriterqueue/all/conanfile.py [hunk 1]
```diff
@@ -11,27 +14,31 @@ class ReaderWriterQueue(ConanFile):
...
+            copy(self, "readerwritercircularbuffer.h", src=self.source_folder, dst=os.path.join("include", "readerwriterqueue"))
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
 
     def package_info(self):
-        if self.settings.os == "Linux":
+        self.cpp_info.bindirs = []
+        self.cpp_info.frameworkdirs = []
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/conanfile.py:40]:
```suggestion
```

File: recipes/readerwriterqueue/all/conanfile.py [hunk 2]
```diff
@@ -11,27 +14,31 @@ class ReaderWriterQueue(ConanFile):
...
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/conanfile.py:42]:
```suggestion
```

File: recipes/readerwriterqueue/all/conanfile.py [hunk 3]
```diff
@@ -11,27 +14,31 @@ class ReaderWriterQueue(ConanFile):
...
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("readerwriterqueue-{}".format(self.version),
-                  self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/conanfile.py:23]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

File: recipes/readerwriterqueue/all/conanfile.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.files import get, copy, rename
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/conanfile.py:2]:
```suggestion
from conan.tools.files import get, copy
```

File: recipes/readerwriterqueue/all/test_v1_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} ../test_package/test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/test_v1_package/CMakeLists.txt:9]:
```suggestion
cmake_minimum_required(VERSION 3.1)
project(test_package)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package)
```

File: recipes/readerwriterqueue/all/test_v1_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class ConcurrentqueueTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/test_v1_package/conanfile.py:6]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/readerwriterqueue/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -1,9 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.8)
```

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/readerwriterqueue/all/conanfile.py [hunk 8]
```diff
@@ -11,27 +16,28 @@ class ReaderWriterQueue(ConanFile):
...
+    def build(self):
+        pass
 
-        self.copy("LICENSE.md", src=self._source_subfolder, dst="licenses")
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "atomicops.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include", "readerwriterqueue"))
+        copy(self, "readerwriterqueue.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include", "readerwriterqueue"))
+        if Version(self.version) >= "1.0.5":
+            copy(self, "readerwritercircularbuffer.h", src=self.source_folder, dst=os.path.join("include", "readerwriterqueue"))
```
Comment (SpaceIm) [inline recipes/readerwriterqueue/all/conanfile.py:34]: dst is incorrect here, related to https://github.com/conan-io/conan-center-index/issues/16651?

Comment (kletozan) [reply recipes/readerwriterqueue/all/conanfile.py:34]: Thanks! I saw it was fixed in https://github.com/conan-io/conan-center-index/pull/16672.

**Pull request merged**